Integrate App Mining with App.co further

aulneau opened the pull request:
### Staging

[Click here](https://app-co-staging-pr-453.herokuapp.com/)

### What this does

This PR address the following issues:
- https://github.com/blockstack/app.co/issues/435
- https://github.com/blockstack/app.co/issues/318
- https://github.com/blockstack/app.co/issues/377
- https://github.com/blockstack/app.co/issues/459
- https://github.com/blockstack/app.co/issues/468
- https://github.com/blockstack/app.co/issues/432
- https://github.com/blockstack/app.co/issues/461
- https://github.com/blockstack/app.co/issues/446 (partially)

This PR changes the homepage category that displays Popular Blockstack apps to pull apps from App Mining, and list them in their ranking order. I've opted to keep the display much the same (not showing lifetime earnings). Currently they are ordered based on month, not based on lifetime earnings. This could result in them changing each month (which I think is good!). We can order by lifetime earnings if the group would prefer that.

I've also opted to include the lifetime earnings on the app card when you click on an app as shown below (screenshot contains staging API data):

![screen shot 2019-02-11 at 10 14 16 am](https://user-images.githubusercontent.com/11803153/52576559-e983cb00-2de5-11e9-8eb8-738b2881f4ff.png)

I've also fixed the CTA modal on the app mining landing page to reflect the context a bit better.

On the App MIning landing page, I've added the functionality to navigate between months for the 'How much can you earn' section:

<img width="1792" alt="screen shot 2019-02-27 at 11 18 10 am" src="https://user-images.githubusercontent.com/11803153/53505210-6f0fa800-3a81-11e9-8b87-e3d609da4e52.png">

hstove commented on common/constants/index.js:
```diff
@@ -1,3 +1,5 @@
 const headerHeight = '88px'
 
-export { headerHeight }
+const API_URL = `https://app-co-api.herokuapp.com/api`
```
I would much prefer to use the `apiServer` provided in Redux - this makes local development much smoother.

aulneau replied:
oh yeah sure thing

aulneau replied:
This has been resolved

markmhx commented:
@aulneau Getting an error when loading staging:

<img width="1460" alt="screen shot 2019-02-12 at 16 23 13" src="https://user-images.githubusercontent.com/28991/52646158-89268380-2ee2-11e9-954a-c76b14c905c6.png">

markmhx commented:
Looking good! My notes per issue addressed below.

---

Per https://github.com/blockstack/app.co/issues/435 ("Add listing of all monthly rankings for App Mining"):

- I like the date toggle in the "How much can you earn?" section but it is quite subtle and easy to overlook. And there doesn't appear to be a section that lists links to each month results page (e.g. https://app.co/mining/january-2019). Is there a way to incorporate those links without adding a new section? Or is a new section listing "rounds" / months the best way to give these discoverability? Those pages feel like orphans still, and I presume we still want people to find them specifically each month and for past reference?

Per https://github.com/blockstack/app.co/issues/318 ("Add component for recent App Mining winners to homepage"):

- Should we increase the limit from 15 to 25 in this "Popular Blockstack Apps" component? More the merrier?
- Should we bump this module to the top, above "Popular Decentralized Apps"? I'm thinking Blockstack apps are more interesting content for the typical visitor, presuming that most are coming via Blockstack channels these days.
- Should we modify the "Popular Decentralized Apps" component to use the same grid format as the other homepage components while we're in here? That would seem most consistent to me, and I don't see much upfront value to the other column data vs. showing more apps for perusal. We could switch to grid and bump the limit to 6 or 9 similar to the other components down the page.
- Likewise, should we bump "Business Tools" down to 6 to keep things symmetrical? It appears to stand at 7+ and only 7 are available.
- I'm seeing Graphite show up several times in the "Popular Blockstack Apps" component – is this simply due to staging data messiness? ([see screenshot](https://user-images.githubusercontent.com/28991/52786475-37553900-305b-11e9-8458-c0e91911af3f.png))
- When cliicking "View all" for "Popular Blockstack Apps", I see a very different order of apps since that page uses tweets to order. Can we modify that page to reflect the same App Mining ordering as the homepage component? And if so, should we swap "Total Earnings" for "Tweets/Week" as a column?

Per https://github.com/blockstack/app.co/issues/446 ("Improve app profiles"):

- Nitpick: Verb is "pay out" I believe instead of "payout" (noun) (appears we need to change this in the App Mining landing page hero, too)
- Should we hide "Total Earnings" in the app profile when it's $0? ([see screenshot](https://user-images.githubusercontent.com/28991/52787012-e34b5400-305c-11e9-89ca-3e4883d69f67.png))

Per https://github.com/blockstack/app.co/issues/377 ("Provide different modal header for subscription CTA"):

- Nitpick: Do we want to use lowercase for the "updates" word in "Subscribe for updates"? (vs. "Subscribe for Updates")? We seem to go back and forth between header capitalization styles across the landing page and may want to settle on one.

aulneau commented:
> * I'm seeing Graphite show up several times in the "Popular Blockstack Apps" component – is this simply due to staging data messiness? ([see screenshot](https://user-images.githubusercontent.com/28991/52786475-37553900-305b-11e9-8458-c0e91911af3f.png))

This is using the staging date. I can see if @hstove can copy the latest real data to staging so we can have it be more realistic.


> * When cliicking "View all" for "Popular Blockstack Apps", I see a very different order of apps since that page uses tweets to order. Can we modify that page to reflect the same App Mining ordering as the homepage component? And if so, should we swap "Total Earnings" for "Tweets/Week" as a column?

This was touched on briefly during the app.co issue review this week. It was concluded that this is a more involved change to the larger organization of how we rank and display all apps on app.co. I'm happy to apply the 'bandage' of doing something unique to this page, but it sounds like we want to have a larger convo around this.

Otherwise everything else looks good, thanks for the review @markmhx!

jeffdomke commented:
Per #435 agree with Mark's comments and happy to contribute some design ideas. The information is there but it is kind of in stealth mode right now.

aulneau commented:
I've played around with some displaying of the previous months and this is what I've landed on:

![screen shot 2019-02-22 at 10 36 31 am](https://user-images.githubusercontent.com/11803153/53256566-d4345980-368d-11e9-9ba9-3351e49252dd.png)

It will stick with you as you scroll the page, and you can quickly click between months to see how apps have moved around.

For mobile I'm not quite set on a path yet, this is what I've got currently:
![screen shot 2019-02-22 at 10 38 10 am](https://user-images.githubusercontent.com/11803153/53256645-f8903600-368d-11e9-958e-969337e0e516.png)

The arrows to navigate between months still exist there, but I think it probably makes sense to add a list or something to navigate them to a page that just shows all the months, such as `https://app.co/mining/months` which doesn't currently exist.

hstove commented:
This looks good, but how will the mobile UI work with lots of months? Horizontal scroll?

jeffdomke commented:
I think the table of months on the side seems a little non-ideal to me in terms of space it occupies and the prominence. I might suggest using the same setup you have on the mobile version on desktop. And use labels `Feb 2019` and put the highlight stroke on the bottom of the tab. Then on mobile you can just hide this entire feature. 

Agree with your thought on the dedicated URL.

jeffdomke commented:
Per the horizontal scroll on desktop you could just go for a second row. That would take us to what... 12 months at least? By that time we can reconsider.

markmhx commented:
I'd echo @hstove and @jeffdomke's concerns but also wondering how the user can access the dedicated monthly pages of rankings? e.g. https://app.co/mining/january-2019

Maybe like a CTA when a month is toggled that says "View full month" or something similar instead of the "Show more apps"?

aulneau commented:
This sounds good, I'll add it as a row above the table with the 3 char month label.

> but also wondering how the user can access the dedicated monthly pages of rankings? e.g. https://app.co/mining/january-2019

I can add a little icon to imply that they can view a full report of the month, or maybe an additional button that says that.

jeffdomke commented:
![screen shot 2019-02-24 at 10 20 07 am](https://user-images.githubusercontent.com/767828/53301250-679c9480-381e-11e9-89cd-fa6d34981bdc.png)

I think to an unfamiliar user the difference between `full report` and `show more` will not be clear. 

![screen shot 2019-02-24 at 10 21 23 am](https://user-images.githubusercontent.com/767828/53301255-771bdd80-381e-11e9-9603-d3f8ef3fd571.png)

To further expand on the last point you have the entity of `Feb 19` referenced in 3 places when they are all in-fact the same. At the least, I would expect the full report CTA to be visually bound to the other two references in some way. 

![screen shot 2019-02-24 at 10 23 05 am](https://user-images.githubusercontent.com/767828/53301265-a7fc1280-381e-11e9-9c56-2aebf1a3e2e8.png)

Seems to be in reverse time order? I'm not sure the abbreviations are helping, seems like it might just be cleaner to spell out `2019`. 

![moving](https://user-images.githubusercontent.com/767828/53301283-d7ab1a80-381e-11e9-8f0a-c848e489fdda.gif)

The tabs shifting seems like a craft nit. 

Larger point, I'm not even sure you need need tabs here. Why can't this be a spaced out, centered, list of months in that same area. I feel that 95% of users will not interact with this, so it can be a subtle thing. And you can play with only revealing the full report CTA also in some subtle way connected to the month in question.

aulneau commented:
@jeffdomke thanks for the feedback!
I've made some changes to request what you've mentioned:

<img width="1126" alt="screen shot 2019-02-25 at 10 46 17 am" src="https://user-images.githubusercontent.com/11803153/53349291-cd565280-38ea-11e9-9a99-c8dda248c1cb.png">

<img width="1131" alt="screen shot 2019-02-25 at 10 46 27 am" src="https://user-images.githubusercontent.com/11803153/53349295-cf201600-38ea-11e9-870c-b0b492d7aa44.png">

I moved away from the tabs to the pills, centered most of the content, and changed the action at the bottom to not show all the items inline, but to link to the full display of apps for the month that is selected.

markmhx commented:
I feel the tension with the time-based ordering base, too, but think placing the default selected tab to the left is more important than ordering time ascending going right. So this looks good to me.

jeffdomke commented:
Changes look good to me. One thing I noticed is that the app profile App Mining payout numbers were not correct, I'm assuming this is due to stale data on staging? For example: 

![screen shot 2019-02-26 at 12 42 23 pm](https://user-images.githubusercontent.com/767828/53434171-feef1c80-39c3-11e9-92b4-b837098aede9.png)

aulneau commented:
@jeffdomke yeah the data on staging is very different/wrong

jeffdomke commented:
Cool, I'm good. @markmhx I think you are the last reviewer here.

markmhx commented:
Looks good, but didn't we want to show more apps in the "Popular Blockstack Apps" section on the homepage? I'd suggest 12 at least.

I'd love to switch the "Popular Decentralized Apps" component to use the same grid format as the other components on the homepage, too, assuming its an easy switch here. I'd show at least six in that format (vs. the five I currently see) since we want to balance it out.

Otherwise, we're good to 🛳!

aulneau commented:
@markmhx the staging is only showing a smaller amount of apps, on prod it will show 23 apps!

I think we should bump the display of that section to a different PR. I'm going to merge this in :)
